src/sentry/api/endpoints/seer_rpc.py

```diff
@@ -160,9 +207,358 @@ def get_organization_autofix_consent(*, org_id: int) -> dict:
+    # We can tolerate potential false positives if downstream uses of this data filter
+    # out irrelevant issues.
+    sentry_filenames.add(pr_filename)
+    sentry_filenames.update(_left_truncated_paths(pr_filename, max_num_left_truncated_paths))
+    return project_set, sentry_filenames
+
+
+def get_issues_related_to_file_patches(
+    *,
+    organization_id: int,
+    provider: str,
+    external_id: str,
+    pr_files: list[PrFile],
+    max_num_issues_per_file: int = MAX_NUM_ISSUES_PER_FILE_DEFAULT,
+) -> dict[str, list[dict[str, Any]]]:
+    """
+    Get the top issues related to each file by looking at matches between functions in the patch
+    and functions in the issue's event's stacktrace.
+
+    Each issue includes its latest serialized event.
+    """
+
+    try:
+        repo = Repository.objects.get(
+            organization_id=organization_id, provider=provider, external_id=external_id
+        )
+    except Repository.DoesNotExist:
+        logger.exception(
+            "Repo doesn't exist",
+            extra={
+                "organization_id": organization_id,
+                "provider": provider,
+                "external_id": external_id,
+            },
+        )
+        return {}
+
+    repo_id = repo.id
+
+    pr_files = safe_for_fetching_issues(pr_files)
+    pullrequest_files = [
+        PullRequestFile(filename=file["filename"], patch=file["patch"]) for file in pr_files
+    ]
+
+    filename_to_issues = {}
+    patch_parsers: dict[str, LanguageParser | SimpleLanguageParser] = PATCH_PARSERS
+
+    for file in pullrequest_files:
+        projects, sentry_filenames = _get_projects_and_filenames_from_source_file(
+            organization_id, repo_id, file.filename
+        )
+        if not projects:
+            logger.error("No projects", extra={"file": file.filename})
+            continue
+
+        file_extension = file.filename.split(".")[-1]
+        if file_extension not in patch_parsers:
+            logger.error("No language parser", extra={"file": file.filename})
+            continue
+        language_parser = patch_parsers[file_extension]
+
+        function_names = language_parser.extract_functions_from_patch(file.patch)
+        if not function_names:
+            logger.error("No function names", extra={"file": file.filename})
+            continue
+
+        issues = get_issues_with_event_details_for_file(
+            list(projects),
+            list(sentry_filenames),
+            list(function_names),
+            max_num_issues_per_file=max_num_issues_per_file,
+        )
+        filename_to_issues[file.filename] = issues
+
+    return filename_to_issues
+
+
+seer_method_registry: dict[str, Callable[..., dict[str, Any]]] = {
     "get_organization_slug": get_organization_slug,
     "get_organization_autofix_consent": get_organization_autofix_consent,
+    "get_issues_related_to_file_patches": get_issues_related_to_file_patches,
```

Comment:
should we have all the logic for this outside in another file and keep this file only specific to seer rpc?

Reply:
discussed. will move to `src/sentry/autofix` and rename that dir to `src/sentry/ai`